hooks/tk-multi-publish2/basic/submit_for_review.py

```diff
@@ -0,0 +1,300 @@
+        """
+        Dictionary defining the settings that this plugin expects to recieve
+        through the settings parameter in the accept, validate, publish and
+        finalize methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts
+        as part of its environment configuration.
+        """
+        return {}
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["maya.*", "file.maya"]
+        """
+        return ["*.sequence"]
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A submit for review task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled. 
+            This plugin makes use of the tk-multi-reviewsubmission app; if this
+            app is not available then the item will not be accepted, this method
+            will return a dictionary with the accepted field that has a value of
+            False.
+
+            Several properties on the item must also be present for it to be
+            accepted. The properties are 'path', 'publish_name', 'color_space',
+            'first_frame' and 'last_frame'
+        """
+
+        accepted = True
+        review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
+        if review_submission_app is None:
+            accepted = False
+            self.logger.warning(
```

Comment:
Since this is setup to match all `*.sequence` items, we should change these to `debug` instead of `warning` messages. A regular nuke write node, for example, will pop up as a warning immediately in the publisher which would be confusing. Sorry I didn't notice this before.

Reply:
changed